backend/src/main/java/ai/verta/modeldb/entities/ArtifactEntity.java

```diff
@@ -51,9 +53,21 @@ public ArtifactEntity(Object entity, String fieldType, Artifact artifact) {
     }
 
     this.field_type = fieldType;
+    var path = artifact.getPath();
     var uploadCompleted = !artifactStoreConfig.getArtifactStoreType().equals(CommonConstants.S3);
     if (artifact.getUploadCompleted()) {
       uploadCompleted = true;
+    } else {
+      path =
+          artifactStoreConfig.storeTypePathPrefix()
+              + entityName
+              + (entityId.map(id -> "/" + id + "/").orElse("/"))
+              + artifact.getKey();
+    }
```

Comment:
Similar concerns

Reply:
Simplify logic and implemented it for nfs as well as per our discussion